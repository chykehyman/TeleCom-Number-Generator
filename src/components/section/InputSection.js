import React, { Fragment } from "react";
import styled from "styled-components";
import { func, string } from "prop-types";

const Input = styled.input`
  outline: none;
  border-radius: 5px;
  height: 40px;
  font-size: 16px;
  box-shadow: 5px 5px 5px #aaaaaa;
  color: #555;
  border: none;
  padding: 10px;
`;

const Button = styled.button`
  border: 1px solid #de3e3e;
  border-radius: 20px;
  color: #de3e3e;
  background-color: white;
  cursor: pointer;
  height: 40px;
  font-weight: bold;
  font-size: 16px;
  width: 150px;
  text-transform: uppercase;
  margin-left: 10px;
  &:hover {
    background-color: #de3e3e;
    color: white;
  }
  margin-right: 10px;
  outline: none;
`;

const Select = styled.select`
  outline: none;
  height: 40px;
  margin-left: 10px;
  cursor: pointer;
`;

const ErrorDiv = styled.div`
  color: red;
  margin: 10px 0 0;
  font-style: italic;
  font-size: 13px;
`;

const Wrapper = styled.div`
  display: flex;
  width: 100%;
  margin-top: 30px;
`;

const InputSection = ({
  handleChange,
  inputValue,
  error,
  generatePhoneNumbers,
  handleSortChange
}) => (
  <Fragment>
    <Wrapper>
      <Input
        type="number"
        placeholder="enter number here..."
        value={inputValue}
        onChange={handleChange}
      />
      <Button onClick={generatePhoneNumbers}>Generate</Button>
      <Select defaultValue="default" onChange={handleSortChange}>
        <option value="default" disabled>
          Sort By
        </option>
        <option value="asc">Ascending</option>
        <option value="desc">Descending</option>
      </Select>
    </Wrapper>
    {error && <ErrorDiv>{error}</ErrorDiv>}
  </Fragment>
);

InputSection.propTypes = {
  handleChange: func.isRequired,
  inputValue: string.isRequired,
  error: string.isRequired,
  generatePhoneNumbers: func.isRequired,
  handleSortChange: func.isRequired
};

export default InputSection;
