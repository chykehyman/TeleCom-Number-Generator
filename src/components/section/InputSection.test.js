import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import InputSection from "./InputSection";

configure({ adapter: new Adapter() });

describe("Test case for <InputSection /> component", () => {
  const props = {
    handleChange: jest.fn(),
    inputValue: "",
    error: "",
    generatePhoneNumbers: jest.fn(),
    handleSortChange: jest.fn()
  };

  it("should render the InputSection component", () => {
    const shallowWrapper = shallow(<InputSection {...props} />);
    expect(shallowWrapper).toMatchSnapshot();
  });

  it("should show error div component when an error is triggered", () => {
    const newProps = {
      ...props,
      error: "An error occurred"
    };
    const shallowWrapper = shallow(<InputSection {...newProps} />);
    expect(shallowWrapper.find('ErrorDiv')).toBeTruthy();
  });
});
