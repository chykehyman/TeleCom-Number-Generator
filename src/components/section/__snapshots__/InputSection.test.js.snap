// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for <InputSection /> component should render the InputSection component 1`] = `
<Fragment>
  <styled.div>
    <styled.input
      onChange={[MockFunction]}
      placeholder="enter number here..."
      type="number"
      value=""
    />
    <styled.button
      onClick={[MockFunction]}
    >
      Generate
    </styled.button>
    <styled.select
      defaultValue="default"
      onChange={[MockFunction]}
    >
      <option
        disabled={true}
        value="default"
      >
        Sort By
      </option>
      <option
        value="asc"
      >
        Ascending
      </option>
      <option
        value="desc"
      >
        Descending
      </option>
    </styled.select>
  </styled.div>
</Fragment>
`;
